infra/src/main/kotlin/fi/fta/geoviite/api/frameconverter/v1/FrameConverterServiceV1.kt

```diff
@@ -81,34 +87,45 @@ constructor(
                 nearby.find { (track, _) -> filterByRequest(track, trackNumbers, request) }
             }
 
-        val trackNumbersAndGeocodingContexts =
-            closestTracks
-                .mapNotNull { it?.track?.trackNumberId }
-                .distinct()
-                .associateWith { trackNumberId ->
-                    trackNumberService.get(layoutContext, trackNumberId) to
-                        geocodingService.getGeocodingContext(layoutContext, trackNumberId)
-                }
-
-        val trackDescriptions = getTrackDescriptions(params, closestTracks.mapNotNull { it?.track })
+        val trackNumberInfo = getTrackNumberInfoForLocationTrackHits(closestTracks, layoutContext)
+        val detailInfo = getLocationTrackFeatureDetailInfo(params, closestTracks.mapNotNull { it?.track })
 
         return closestTracks
             .mapIndexed { index, trackHit ->
                 val (request, _) = requestsWithPoints[index]
                 if (trackHit == null) createErrorResponse(request.identifier, FrameConverterErrorV1.FeaturesNotFound)
                 else {
+                    val trackDetailInfo = detailInfo?.get(trackHit.track.id)
                     calculateCoordinateToTrackAddressResponse(
                         request,
                         trackHit,
-                        trackDescriptions?.get(trackHit.track.id),
+                        trackDetailInfo,
                         params,
-                        trackNumbersAndGeocodingContexts,
+                        trackNumberInfo,
                     )
                 }
             }
             .toList()
     }
 
+    private fun getTrackNumberInfoForLocationTrackHits(
+        closestTracks: List<LocationTrackCacheHit?>,
```

Comment:
Mietin että olisikohan tämä rajapinnan puolesta selkeämpi jos sen sijasta että tämä ottaisi listan kakkuosumia, niin se ottaisikin vain listan `LocationTrack`-olioita 🤔 Tällöin käytännössä kutsujan harteille jäisi melkein kokonaan tuo rivien 115-116 `closestTracks.mapNotNull { ... }`, mutta sillä saisi mm. korvattua funktion nimestä `LocationTrackHits`-termin yleisemmin käytetyllä `LocationTracks`. Huomasin itse tuolla ylempänä tämän kutsukohtaan törmätessäni tuon `Hits`-termin sekoittavan itseäni jonkin verran.